src/audio/host.c

```diff
@@ -350,6 +349,13 @@ static void host_update_position(struct comp_dev *dev, uint32_t bytes)
 	if (!hd->no_stream_position) {
 		hd->report_pos += bytes;
 
+		/* update position every period to support host query to
+		 * get the latest value.
+		 */
+		pipeline_get_timestamp(dev->pipeline, dev, &hd->posn);
+		mailbox_stream_write(dev->pipeline->posn_offset,
+				     &hd->posn, sizeof(hd->posn));
+
```

Comment:
@yaochunhung this is a good fix, but it looks like we can tune it by setting the hd->report_pos to align with the 1ms data size. This is probably something the driver or topology should ask for (as it has impact on power) so could be fine tuned per use case.

Reply:
@lgirdwood  Thanks for your comment. I think this change will update data based on pipeline' period. and it will be set by different use cases(via topology). It will increase some MCPS due to updating host position information(maybe also impact power). Please advice if my opinion is incorrect. Thanks.

Reply:
yes, I think if we can set the update period via teh driver or topology would be the best way forward. I dont think it will have too much impact on MCPS though, but it could impact power (on Intel and maybe others) as it can write to (and wake) IP blocks.

Reply:
@lgirdwood  Does it mean that we need to define another parameter for updated_period?
I have little bit confusion about power consumption because only pipeline_get_timestamp and mailbox_stream_write without sending ipc to AP side(send ipc is still under checking by host_period_bytes). Will it impact intel platforms? Thanks.

Reply:
Sorry @yaochunhung I misread the reply. Yes we need a new flag coming from the driver i.e.
```c
/* PCM params info - SOF_IPC_STREAM_PCM_PARAMS */
struct sof_ipc_pcm_params {
	struct sof_ipc_cmd_hdr hdr;
	uint32_t comp_id;
	uint32_t flags;		/**< generic PCM flags - SOF_PCM_FLAG_ */ <<<<<<<<<< Can we ad a new flag here for continuous updates
	uint32_t reserved[2];
	struct sof_ipc_stream_params params;
} __attribute__((packed, aligned(4)));
```

Reply:
@lgirdwood  Thanks.
I check sof_ipc_pcm_params is filled in this function sof_pcm_hw_params at linux side and without platform specific callback. Not sure if what you want is to use sof_ipc_stream_params. The sof_ipc_stream_params could be add a flag naming as continuous_update(see code below maybe a better name for it) and it also can be decided in the pcm_hw_params callback function by different platforms.
```c
struct sof_ipc_stream_params {
	struct sof_ipc_hdr hdr;
	struct sof_ipc_host_buffer buffer;
	uint32_t direction;	/**< enum sof_ipc_stream_direction */
	uint32_t frame_fmt;	/**< enum sof_ipc_frame */
	uint32_t buffer_fmt;	/**< enum sof_ipc_buffer_format */
	uint32_t rate;
	uint16_t stream_tag;
	uint16_t channels;
	uint16_t sample_valid_bytes;
	uint16_t sample_container_bytes;

	uint32_t host_period_bytes;
	uint16_t no_stream_position; /**< 1 means don't send stream position */
        uint16_t continuous_update_position; /**< 1 means continue updating stream position */  <<<<< add a flag for it
	uint16_t reserved[2];
	uint16_t chmap[SOF_IPC_MAX_CHANNELS];	/**< channel map - SOF_CHMAP_ */
} __packed;
```

Reply:
@lgirdwood  Thanks for suggestion. I add a PR to add this flag in kernel first. https://github.com/thesofproject/linux/pull/3353/ Please help to check if it is ok. Thanks!

Reply:
@lgirdwood  Do I need to modify abi.h because it looks like there is a checkpatch warning as below? If yes, how to modify it? Thanks. 
https://sof-ci.01.org/sofpr/PR5137/build11487/checkpatch/
```
WARNING: Please update ABI in accordance with http://semver.org
#72: FILE: src/include/ipc/stream.h:97:
 	uint16_t no_stream_position; /**< 1 means don't send stream position */
+	uint16_t cont_update_posn; /**< 1 means continue update stream position */
```

Reply:
yes, we need to label the PR with the ABI flag - that way everyone knows there is ABI change. I've marked both PRs with ABI label now.

Reply:
@lgirdwood  I also update abi version to ABI-3.21. thanks